lblod/ember-rdfa-editor-lblod-plugins
# Snippet RDFa linking

### Overview
Adds a new sample page which is set up for editing snippets. The main addition here is to enable `hasResourceImports` on the doc node of the editor schema.

~~This is a draft as there are a few issues to work out, which will be listed in the how to test section.~~

##### connected issues and PRs:
Requires linking to (and includes in the package.json) https://github.com/lblod/ember-rdfa-editor/pull/1212 which adds most of the capabilities used.
Can be linked to https://github.com/lblod/ember-rdfa-editor/pull/1213 to try out creating links *to* imported resources instead of just *from* them.
Used by https://github.com/lblod/frontend-reglementaire-bijlage/pull/263
Jira ticket: [GN-4898](https://binnenland.atlassian.net/browse/GN-4898)

### Setup
Needs some hacking to get a snippet with the required imported resources specs into the insert list.

### How to test/reproduce
There are 3 steps to this:
1. Create a snippet with imported resources:
	- Go to the snippet edit sample page
	- Insert the snippetTemplate template, this contains the attribute on the doc node that imports some resources to the editor
	- Create arbitrary RDFa in the editor
	- Click on the 'document' node (so click on the page outside of the content), and verify that the RDFa tools give the type 'document'
	- Open the 'Add Relationship' modal, create a link as normal, except also specify the subject. For the first imported resource you'll need to enter the placeholder URI into the power-select 'search' and hit enter, this will set the field to this new value. Later you can select an existing import or create a new one in the same way.
	- If testing the experimental editor branch, select a resource in the editor and click 'add relationship'. Select the type as 'NamedNode' and manually type the imported resource URI.
	- export the HTML from the editor as you'll need it later.
2. Create a template containing a snippet placeholder:
	- Go to the regulatory statements sample page
	- Insert a snippet placeholder (for now any snippet list selected will have the imported resources added that match those in the sample template)
	- Select the placeholder and see that the RDFa relationship editor has a new section listing the imported resources. You can click on the `+` to choose the 'real' resource URI that it links to in the document.
3. Create a real document from a template with a snippet placeholder:
	- Select the placeholder and click 'insert snippet' from the sidebar
	- Unfortunately to get a snippet with the right imported resources, you'll need to override the snippet-plugin utils `fetch-data` function `fetchSnippets` to return the html you created in 1.
	- You should see the snippet contents now and if you click within the snippet, the RDFa tools should show the correct links to the outer document.
	- You can also add more snippets using the new tooling, these should also be linked.

Known issues:
- ~Links from the outer document to the snippet are not set up, the page needs to be refreshed before they're wired up correctly.~
- ~Still some hacks in place.~
- Tools are not pretty or easy to see the required data.
- When editing a link from an imported resource, the subject isn't populated in the modal.

### Challenges/uncertainties
Many.

### Checks PR readiness
